cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/Language.kt

```diff
@@ -111,6 +119,53 @@ abstract class Language<T : LanguageFrontend> : Node() {
     init {
         this.also { this.language = it }
     }
+
+    /**
+     * Determines how to propagate types across binary operations since these may differ among the
+     * programming languages.
+     */
+    open fun propagateTypeOfBinaryOperation(operation: BinaryOperator): Type {
+        if (operation.operatorCode == "==" || operation.operatorCode == "===") {
+            // A comparison, so we return the type "boolean"
+            return this.simpleTypes.values.firstOrNull { it is BooleanType }
+                ?: this.simpleTypes.values.firstOrNull { it.name.localName.startsWith("bool") }
```

Comment:
Probably just return UnknownType in this case, everything else is a little bit of a weird guess work I would say.

Reply:
Also in the second case looking for a name starting with `"bool"`? The BooleanType exists but it is almost never used. So, if we want to get rid of this check, we have use that type in all languages and not only in go

Reply:
> Also in the second case looking for a name starting with `"bool"`? The BooleanType exists but it is almost never used. So, if we want to get rid of this check, we have use that type in all languages and not only in go

Well, we should ;)

Reply:
I can quickly whip up a PR that does that for the supported languages.

Reply:
See. https://github.com/Fraunhofer-AISEC/cpg/pull/1113.

After some consideration we can probably leave the look for `boolXXX` part in, because not all languages necessarily have a "real" boolean type. Although its mainly C and python and I guess that we need to implement this particular function for those languages anyway since they are crazy?